rotations/demonhunter/vengeance.lua

```diff
@@ -28,30 +24,31 @@ local exeOnLoad = function()
 end
 
 local _All = {
-	-- keybind
-  { "Infernal Strike", "keybind(lalt)" },
-  { "Sigil of Flame", "keybind(lcontrol)" },
-
-	{ "/stopcasting\n/stopattack\n/cleartarget\n/stopattack\n/cleartarget", { "player.combat.time >= 300", (function() return F('dpstest') end) }},
+	-- Keybinds
+	{ "Infernal Strike", "keybind(lalt)" },
+	{ "Sigil of Flame", "keybind(lcontrol)" },
 }
 
 local _Cooldowns = {
-  { "Lifeblood" },
-  { "Berserking" },
-  { "Blood Fury" },
+	--{ "Lifeblood" }, -- No point for this. Only Blood and Night Elves can be Demon Hunters
+	--{ "Berserking" }, -- No point for this. Only Blood and Night Elves can be Demon Hunters
+	--{ "Blood Fury" }, -- No point for this. Only Blood and Night Elves can be Demon Hunters
 }
 
 local _Survival = {
-	{ "Fracture", { "player.pain >= 80", "player.buff(Soul Fragments).count >= 5", "player.pain >= 80" }},
 	{ "Soul Cleave", "player.pain >= 80" },
-	{ "Fel Devastation", "player.health <= 75" },
-	{ "Metamorphosis", { "!player.buff(Demon Spikes)", "!target.debuff(Fiery Brand)", "!player.buff(Metamorphosis)", "player.health <= 70" }},
+	{ "Soul Barrier", { "talent(7,3)", "player.health <= 60" }},
```

Comment:
The engine already checks if the spell is provided via the talent selection and it's not really necessary to specifically check for the talent of the spell you're trying to cast.  The talent checks are good for example when you want to take some action when some certain talent is not taken.